modules/platforms/dotnet/Apache.Ignite.Core.Tests/Cache/CacheAbstractTransactionalTest.cs

```diff
@@ -862,6 +862,50 @@ public void TestTransactionScopeAllOperations()
             }
         }
 
+        /// <summary>
+        /// Tests that read operations lock keys in Serializable mode.
+        /// </summary>
+        [Test]
+        public void TestTransactionScopeWithSerializableIsolationLocksKeysOnRead()
+        {
+            Action<Func<ICache<int, int>, int, int>>
+                test = TestTransactionScopeWithSerializableIsolationLocksKeysOnRead;
+
+            test((cache, key) => cache[key]);
```

Comment:
Can ```ContainsKey``` and ```ContainsKeys``` trigger transnational behavior?

Reply:
Double checked this - no, `ContainsKey` and `ContainsKeys` do not participate in transactions.